compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -11885,18 +11896,20 @@ OMR::Z::TreeEvaluator::arraycmplenEvaluator(TR::Node * node, TR::CodeGenerator *
    TR::RegisterPair * firstPair = cg->allocateConsecutiveRegisterPair(firstLen, firstBaseReg);
    TR::RegisterPair * secondPair = cg->allocateConsecutiveRegisterPair(secondLen, secondBaseReg);
    TR::Register * resultReg = cg->allocateRegister();
+   TR::LabelSymbol * shortCircuitLabel = generateLabelSymbol(cg);
    TR::Instruction * cursor;
 
    TR::RegisterDependencyConditions * dependencies = cg->createDepsForRRMemoryInstructions(node, firstPair, secondPair);
 
+   generateRRInstruction(cg, TR::InstOpCode::getLoadRegOpCode(), node, resultReg, orgLen);
+   generateS390CompareAndBranchInstruction(cg, TR::InstOpCode::CLR, node, firstBaseReg, secondBaseReg, TR::InstOpCode::COND_BE, shortCircuitLabel);
```

Comment:
Similar comment as above, baseRegs would be 64-Bit on 64-Bit platform and would be 32-bit on 32-bit platform. 

You can use the query https://github.com/eclipse/omr/blob/83cb59850cf7c9da28183aeeaa7747908160835a/compiler/z/codegen/InstOpCode.cpp#L458 to automatically get the instOpCode that is appropriate for the platform.